File: app/src/main/java/com/mindorks/bootcamp/learndagger/ui/HomeFragment.java
package com.mindorks.bootcamp.learndagger.ui;

import android.app.Application;
import android.app.Fragment;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.mindorks.bootcamp.learndagger.R;

import com.mindorks.bootcamp.learndagger.MyApplication;
import com.mindorks.bootcamp.learndagger.di.component.DaggerFragmentComponent;
import com.mindorks.bootcamp.learndagger.di.module.FragmentModule;

import javax.inject.Inject;



/**
 * Created by admin on 5/9/2019.
 */

public class HomeFragment extends Fragment {

     @Inject
     HomeViewModel homeViewModel;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getDependencies();
    }

    private void getDependencies() {
        DaggerFragmentComponent
                .builder()
                .applicationComponent(((MyApplication) getActivity().getApplication()).applicationComponent)
                .fragmentModule(new FragmentModule(this))
                .build()
                .inject(this);


    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_home, container, false);

        return view;
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);


        TextView tvData = getView().findViewById(R.id.tvData);

            tvData.setText(homeViewModel.getSomeData()+" Network Available:-"+homeViewModel.isNetworkAvailable());


    }
}

File: app/src/main/java/com/mindorks/bootcamp/learndagger/ui/HomeFragment.kt
package com.mindorks.bootcamp.learndagger.ui


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.mindorks.bootcamp.learndagger.MyApplication
import com.mindorks.bootcamp.learndagger.R
import com.mindorks.bootcamp.learndagger.di.component.DaggerFragmentComponent
import com.mindorks.bootcamp.learndagger.di.module.FragmentModule
import javax.inject.Inject

class HomeFragment: Fragment(){


    @set:Inject
     internal var  homeViewModel: HomeViewModel? = null

    val TAG = "HomeFragment"

    fun newInstance():Fragment{
        return HomeFragment()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getDependencies()
    }

    private fun getDependencies() {
        val aboutComponent = DaggerFragmentComponent.builder()
              //  .applicationComponent((application as MyApplication).applicationComponent)
                .applicationComponent((context!!.applicationContext as MyApplication).applicationComponent)
                .fragmentModule(FragmentModule(this))
                .build()

        aboutComponent.inject(this)
    }

   /* override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View {
        var rootView = inflater!!.inflate(R.layout.fragment_home, container, false)
        return rootView

    }*/

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        var rootView = inflater!!.inflate(R.layout.fragment_home, container, false)
        return rootView
    }


    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        val tvData = view!!.findViewById<TextView>(R.id.tvData) as TextView

        tvData.setText(homeViewModel?.getSomeData())
    }

}